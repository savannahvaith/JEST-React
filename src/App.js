import logo from './Resources/logo.svg';
import './Resources/App.css';
import ComponentWithProps from './Components/ComponentWithProps';
import Form from './Components/Form';

function App() {
  const props = {
    headerText: `A header`,
    contentProp: `Some Content`,
    numericProp: 100,
    valueProp: 10000,
    expressionProp: 2 + 2,
    functionProp: () => `hello world`,
    objectProp: {
      key1: `key1 value`,
      key2: 2,
      key3: [`key3`, `array`, `values`]
    },
    arrayProp: [1, 2, 3]
  }
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn how to write in React
        </a>
      </header>
      {/* <ComponentWithProps {...props}/> */}
      <Form/>
    </div>
  );
}

export default App;
