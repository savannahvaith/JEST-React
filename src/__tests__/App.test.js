import { render, screen } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom';
import {create} from 'react-test-renderer';

// add a jest mock function 
// create my own implementation of a component to override te default
jest.mock(`../Components/ComponentWithProps`, () => {
  return function ComponentWithPropsMock(props){
    return(<p>Mocked componentswithprops</p>)
  }
});

test.skip('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

// snapshot test 
describe(`Snapshot test of App`, () => {

  test(` it should match the snapshot every time it is rendered`, () => {
    const instance = create(<App/>).toJSON(); // it returns an object representting the rendered tree. 
    expect(instance).toMatchSnapshot();
    // it will create a snapshotfor the first time if it doesn't already exists
  });
});