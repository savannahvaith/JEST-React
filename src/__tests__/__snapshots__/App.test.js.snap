// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of App  it should match the snapshot every time it is rendered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn how to write in React
    </a>
  </header>
  <p>
    Mocked componentswithprops
  </p>
</div>
`;
